spec/requests/posts_spec.rb

```diff
@@ -0,0 +1,26 @@
+require 'rails_helper'
+
+describe 'GET all posts for an user' do
```

Comment:
- [ ] As this is a request test, Kindly sure that you use `type: request` as shown in the example below or use `config.infer_spec_type_from_file_location!` in the Rspec config file (in that case just placing a file in the spec/requests dir will be enough).

```ruby
 require 'rails_helper'

 RSpec.describe "Users", type: :request do
   ...
 end
```